src/components/DayOverview/DayOverviewPdf.tsx

```diff
@@ -0,0 +1,204 @@
+    cell_logo: { flex: 1 },
+    cell_name: { flex: 4 },
+    cell_postcode: { flex: 4 },
+    cell_time: { flex: 2 },
+    cell_collection: { flex: 4 },
+    cell_instructions: { flex: 10 },
+});
+
+const getCurrentDate = (date: Date): string => {
+    const year = date.getFullYear();
+    const month = (date.getMonth() + 1).toString().padStart(2, "0");
+    const day = date.getDate().toString().padStart(2, "0");
+
+    return `${year}-${month}-${day}`;
+};
+
+const dateTimeToAMPM = (datetime: string): string => {
+    return new Date(datetime).getHours() <= 11 ? "AM" : "PM";
+};
+
+// IconDefinition[4] is defined as string | string[] but in reality there is no use case
+// with string[] as svgPathData
+const CustomSVG: React.FC<CustomSVGProps> = ({ icon, color, fill }) => (
+    <Svg width={11} height={11} viewBox="0 0 512 512" style={styles.svg}>
+        <Path
+            fill={fill ? color : undefined}
+            stroke={color}
+            strokeWidth={40}
+            d={icon.icon[4] as string}
+        />
+    </Svg>
+);
+
+const DayOverviewMargin: React.FC<{}> = () => {
+    return <View style={styles.margin} fixed={true}></View>;
+};
+
+const DayOverviewDate: React.FC<DayOverviewDateProps> = ({ date }) => {
+    const dateString = getCurrentDate(date);
+    return (
+        <Text>
+            <Text style={styles.bold}>Date: </Text>
+            {dateString}
+        </Text>
+    );
+};
+
+const DayOverviewLocation: React.FC<DayOverviewLocationProps> = ({ location }) => (
+    <Text>
+        <Text style={styles.bold}>Location: </Text>
+        {location}
+    </Text>
+);
+
+const DayOverviewHeader: React.FC<{}> = () => {
+    return (
+        <View style={[styles.row, styles.bold, { borderTop: "1 solid black" }]} fixed={true}>
+            <View style={[styles.cell_logo, styles.cell]}></View>
+            <View style={[styles.cell_name, styles.cell]}>
+                <Text>Name</Text>
+            </View>
+            <View style={[styles.cell_postcode, styles.cell]}>
+                <Text>Postcode</Text>
+            </View>
+            <View style={[styles.cell_time, styles.cell]}>
+                <Text>Time</Text>
+            </View>
+            <View style={[styles.cell_instructions, styles.cell]}>
+                <Text>Instructions</Text>
+            </View>
+        </View>
+    );
+};
+
+const DayOverviewRow: React.FC<DayOverviewRowProps> = ({ row }) => {
+    return (
+        <View style={styles.row} wrap={false}>
+            <View style={[styles.cell_logo, styles.cell]}>
+                <View style={styles.row}>
+                    <CustomSVG icon={faSquare} color="black" fill={false} />
+                    {row.clients!.flagged_for_attention && (
+                        <CustomSVG icon={faFlag} color="orange" fill={true} />
+                    )}
+                </View>
+            </View>
+            <View style={[styles.cell_name, styles.cell]}>
+                <Text>{row.clients!.full_name}</Text>
```

Comment:
In parcels table, the setting ensures `client_id` must be a foreign key mapping to client table's primary id. Hence, when I do a join function to client, there must be an existing client and the info cannot be null.
However, typescript does not know it and will complain client might possibly be null, which is impossible.

Also, for the `collection_date`, since a range filter is used, it also cannot be null, hence there is a ! too